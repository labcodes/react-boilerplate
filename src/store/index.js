import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';

import actions from '../actions';

// ====

const store = createStore(
    actions,
    applyMiddleware(logger)
);

// ====

export default store;
