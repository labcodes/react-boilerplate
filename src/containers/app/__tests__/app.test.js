import React from 'react';
import { shallow } from 'enzyme';

// Test helpers
import renderer from 'react-test-renderer';

// Containers connected
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from 'store';

// Component
import App from 'containers/app';

describe('App [Snapshot]', () => {
  it('Renders correctly', () => {
    const component = (
      <BrowserRouter>
        <Provider store={store}>
          <App />
        </Provider>
      </BrowserRouter>
    );

    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render the message [User is logged].', () => {
    const component = shallow(
      <BrowserRouter>
        <Provider store={store}>
          <App logged={false} />
        </Provider>
      </BrowserRouter>
    );
    // const paragraph = component.children().find('p');
    // console.warn(paragraph.text())
    // expect(paragraph.text()).toEqual('User is logged');
  });
});
