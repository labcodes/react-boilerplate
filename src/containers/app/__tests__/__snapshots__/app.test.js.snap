// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App [Snapshot] Renders correctly 1`] = `
<section
  className="page-wrapper "
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 150ms ease-in-out",
    }
  }
>
  <div
    className="sc-bxivhb EYjjH"
  >
    <header
      className="sc-bdVaJa eoCrfV"
    >
      <img
        alt="logo"
        className="sc-htpNat iFrFVN"
        src="logo.svg"
      />
      <h1
        className="sc-bwzfXH dullrb"
      >
        Welcome to React
      </h1>
      <button
        onClick={[Function]}
      >
        change user status
      </button>
    </header>
    <p
      className="sc-ifAKCX bsUOqa"
    >
      User is unlogged.
    </p>
  </div>
</section>
`;
